gateway/middleware.go

```diff
@@ -0,0 +1,53 @@
+package gateway
+
+import (
+	"log"
```

Comment:
I think https://pkg.go.dev/github.com/go-kit/log is a better option, cortex uses it, it lets you set up warnings and json logs, it's nice to have this from the beginning